src/optimagic/optimizers/iminuit_migrad.py

```diff
@@ -0,0 +1,154 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+from numpy.typing import NDArray
+
+from optimagic import mark
+from optimagic.config import IS_IMINUIT_INSTALLED
+from optimagic.optimization.algo_options import (
+    STOPPING_MAXFUN,
+    STOPPING_MAXITER,
+)
+from optimagic.optimization.algorithm import Algorithm, InternalOptimizeResult
+from optimagic.optimization.internal_optimization_problem import (
+    InternalOptimizationProblem,
+)
+from optimagic.typing import AggregationLevel
+
+if IS_IMINUIT_INSTALLED:
+    from iminuit import Minuit
+
+
+@mark.minimizer(
+    name="iminuit_migrad",
+    solver_type=AggregationLevel.SCALAR,
+    is_available=IS_IMINUIT_INSTALLED,
+    is_global=False,
+    needs_jac=True,
+    needs_hess=False,
+    supports_parallelism=False,
+    supports_bounds=True,
+    supports_linear_constraints=False,
+    supports_nonlinear_constraints=False,
+    disable_history=False,
+)
+@dataclass(frozen=True)
+class IminuitMigrad(Algorithm):
+    stopping_maxfun: int = STOPPING_MAXFUN
+    stopping_maxiter: int = STOPPING_MAXITER
+
+    def _solve_internal_problem(
+        self, problem: InternalOptimizationProblem, params: NDArray[np.float64]
+    ) -> InternalOptimizeResult:
+        def wrapped_objective(x: NDArray[np.float64]) -> float:
+            return float(problem.fun(x))
+
+        m = Minuit(wrapped_objective, params, grad=problem.jac)
+
+        bounds = _convert_bounds_to_minuit_limits(
+            problem.bounds.lower, problem.bounds.upper
+        )
+        _set_minuit_limits(m, bounds)
+
+        m.migrad(
+            ncall=self.stopping_maxfun,
+            iterate=self.stopping_maxiter,  # review
+        )
+
+        res = _process_minuit_result(m)
+        return res
+
+
+def _process_minuit_result(minuit_result: Minuit) -> InternalOptimizeResult:
+    """Convert iminuit result to Optimagic's internal result format."""
+
+    res = InternalOptimizeResult(
+        x=np.array(minuit_result.values),
+        fun=minuit_result.fval,
+        success=minuit_result.valid,
+        message=repr(minuit_result.fmin),
+        n_fun_evals=minuit_result.nfcn,
+        n_jac_evals=minuit_result.ngrad,
+        n_hess_evals=None,
+        n_iterations=minuit_result.nfcn,
+        status=None,
+        jac=None,
+        hess=None,
+        hess_inv=np.array(minuit_result.covariance),
+        max_constraint_violation=None,
+        info=None,
+        history=None,
+    )
+    return res
+
+
+def _convert_bounds_to_minuit_limits(
```

Comment:
This function needs a few unit tests. Please add them in a new file: test/optimagic/optimizers/test_iminuit_migrad.py`

Reply:
Sounds good. I'll have a look once you commit.